test/integration/peers.integration.js

```diff
@@ -204,8 +202,76 @@ function recreateDatabases (done) {
 	});
 }
 
+function registerDelegatesUsingHttpRequests (done) {
+	var enableForgingPromises = [];
+	testNodeConfigs.forEach(function (testNodeConfig) {
+		testNodeConfig.secrets.forEach(function (keys) {
+			var enableForgingPromise = popsicle.post({
+				url: 'http://' + testNodeConfig.ip + ':' + (testNodeConfig.port - 1000) + '/api/delegates/forging/enable',
+				headers: {
+					'Accept': 'application/json',
+					'Content-Type': 'application/x-www-form-urlencoded'
+				},
+				body: {
+					key: 'elephant tree paris dragon chair galaxy',
+					publicKey: keys.publicKey
+				}
+			});
+			enableForgingPromises.push(enableForgingPromise);
+		});
+	});
+	Promise.all(enableForgingPromises).then(function () {
+		done();
+	}).catch(function () {
+		done('Error during registering delegates using http requests');
+	});
+}
+
+function waitForAllNodesToBeReady (done) {
+	var nodesReadyCnt = 0;
+	var nodeReadyCb = function (err) {
+		if (err) {
+			return done(err);
+		}
+		nodesReadyCnt += 1;
+		if (nodesReadyCnt === testNodeConfigs.length) {
+			done();
+		}
+	};
+
+	testNodeConfigs.forEach(function (testNodeConfig) {
+		waitUntilBlockchainReady(nodeReadyCb, 20, 2000, 'http://' + testNodeConfig.ip + ':' + (testNodeConfig.port - 1000));
+	});
+}
+
+function establishWSConnectionsToNodes (sockets, done) {
+	var connectedTo = 0;
+	var wampClient = new WAMPClient();
+	//ToDo: find a better way for waiting until all test node being able to receive connections
+	setTimeout(function () {
+		testNodeConfigs.forEach(function (testNodeConfig) {
+			monitorWSClient.port = testNodeConfig.port;
+			var socket = scClient.connect(monitorWSClient);
+			wampClient.upgradeToWAMP(socket);
+			socket.on('connect', function () {
+				sockets.push(socket);
+				connectedTo += 1;
+				if (connectedTo === testNodeConfigs.length) {
+					done();
+				}
+			});
+			socket.on('error', function (err) {});
+			socket.on('connectAbort', function (err) {
+				done('Unable to establish WS connection with ' + testNodeConfig.ip + ':' + testNodeConfig.port);
+			});
+		}, 1000);
```

Comment:
that `1000` should be one line below (expected: `setTimeout(function () {...}, 1000);` now `testNodeConfigs.forEach(function () {...}, 1000);`